nsuns-server/src/auth/middleware.rs

```diff
@@ -0,0 +1,58 @@
+use anyhow::anyhow;
+use axum::{
+    extract::State,
+    middleware::Next,
+    response::{IntoResponse, Redirect, Response},
+};
+use chrono::Utc;
+use http::{Request, StatusCode, Uri};
+use tower_cookies::Cookies;
+
+use crate::error::{ErrorWithStatus, OperationResult};
+
+use super::token::{create_empty_cookie, JwtKeys, COOKIE_NAME};
+
+pub async fn redirect_on_missing_auth_cookie<B>(
+    cookies: Cookies,
+    uri: Uri,
+    request: Request<B>,
+    next: Next<B>,
+) -> Response {
+    if cookies.get(COOKIE_NAME).is_none() && uri.path() != "/login" {
+        Redirect::to("/login").into_response()
```

Comment:
In may be worthwhile adding a `?next=` query param to redirect after logging in.